### PR TITLE
feat(local-configuration): added defaultCompositeDeletePolicy property

### PR DESCRIPTION
**Pull request by birapjr**
### Description of your changes

Added local configuration `defaultCompositeDeletePolicy` property.

Fixes https://github.com/crossplane-contrib/x-generation/issues/41

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

The MR can be tested adding `defaultCompositeDeletePolicy: Foreground` to a generate.yaml and execute the gerenation and the `defaultCompositeDeletePolicy: Foreground` should be added to the output xrd definition.yaml.

gerenate.yaml
```yaml
group: dms.aws.example
name: Endpoint
version: v1beta1
provider:
  baseURL: https://raw.githubusercontent.com/upbound/%s/%s/package/crds/%s
  name: provider-aws
  version: main
  crd:
    file: dms.aws.upbound.io_endpoints.yaml
    version: v1beta1
patchExternalName: false
defaultCompositeDeletePolicy: Foreground
compositions:
  - name: compositeendpoint.dms.aws.example
    provider: example
    default: true
```

expected definition.yaml
```yaml
apiVersion: apiextensions.crossplane.io/v1
kind: CompositeResourceDefinition
metadata:
  name: compositeendpoints.dms.aws.example
spec:
  defaultCompositeDeletePolicy: Foreground
  claimNames:
    kind: Endpoint
    plural: endpoints
  defaultCompositionRef:
    name: compositeendpoint.dms.aws.example
  group: dms.aws.example
```

### CODE REVIEW

File: pkg/main_test.go [hunk 1]
```diff
@@ -2703,3 +2703,295 @@ func Test_noDefaultPatch(t *testing.T) {
...
+
+		var newCRD cv1.CompositeResourceDefinition
+			err = yaml.Unmarshal(y, &newCRD)
+
+		if err != nil {
+			t.Errorf("could not parse definition.yaml file")
+		}
+
+		// assert that the default composite delete policy is set to Background
+		if newCRD.Spec.DefaultCompositeDeletePolicy != nil {
```
Comment (akesser) [inline pkg/main_test.go:2977]: Please also test that the policy is not nil.

File: pkg/main_test.go [hunk 2]
```diff
@@ -2703,3 +2703,295 @@ func Test_noDefaultPatch(t *testing.T) {
...
+			t.Error("could not delete composition file")
+		}
+		err = os.Remove(tempDir)
+		if err != nil {
+			t.Error("could not delete temp directory")
+		}
+	})
+}
+
+func Test_setDefaultCompositeDeletePolicy_Background(t *testing.T) {
```
Comment (akesser) [inline pkg/main_test.go:2853]: Can you combine both test functions? The only difference is the deletion policy. You could use an array with the input and expected values and loop over it calling the function on each. In the Readme you wrote that the default is Background. Please test this also

File: pkg/main.go [hunk 3]
```diff
@@ -655,7 +712,7 @@ func parseArgs(configFile, generatorFile, inputPath, scriptFile, scriptPath, out
 	if err != nil {
 		return err
 	}
-
+
```
Comment (akesser) [inline pkg/main.go:715]: Please remove space

**Review by akesser — CHANGES_REQUESTED** [review]

File: pkg/main_test.go (hunk 2 above)
Comment (birapjr) [reply pkg/main_test.go]: I removed the default value from the README as its done after by [crossplane](https://docs.crossplane.io/latest/concepts/composite-resource-definitions/#defaultcompositedeletepolicy), not possible to test at this point.

File: pkg/main_test.go (hunk 1 above)
Comment (birapjr) [reply pkg/main_test.go]: done at 97fb87b7486f9b7eb0933b049b06bfc860cf6036

File: pkg/main_test.go (hunk 2 above)
Comment (birapjr) [reply pkg/main_test.go]: tests combine done at 97fb87b7486f9b7eb0933b049b06bfc860cf6036

File: pkg/main.go (hunk 3 above)
Comment (birapjr) [reply pkg/main.go]: done at 97fb87b7486f9b7eb0933b049b06bfc860cf6036

**Comment by akesser** [comment]
LGTM
